### ISSUE TITLE
[openldap] memberOf overlay does not work

### ISSUE THREAD

**Issue by mohsensamiei** [bitnami/containers#982]
### Name and Version

bitnami/openldap:2.6.3

### What steps will reproduce the bug?

Hey
Hi have setup openldap with this docker-compose 
```yaml
version: "3.9"

volumes:
  openldap_data:

services:
  openldap:
    image: bitnami/openldap:2
    ports:
      - 1389:1389
      - 1636:1636
    environment:
      - LDAP_ROOT=dc=example,dc=com
      - LDAP_ADMIN_USERNAME=admin
      - LDAP_ADMIN_PASSWORD=adminpassword
    volumes:
      - openldap_data:/bitnami/openldap
```
So execute this:
```
dn: ou=groups,dc=example,dc=com
objectclass: organizationalUnit
ou: groups

dn: ou=users,dc= example,dc=com
objectclass: organizationalUnit
ou: users

dn: cn=user01,ou=users,dc= example,dc=com
cn: user01,
objectclass: inetOrgPerson
objectclass: top
sn: bar01
uid: user01

dn: cn=group01,ou=groups,dc= example,dc=com
cn: group01
member: cn= user01,ou=users,dc= example,dc=com
objectclass: groupOfNames
objectclass: top

dn: cn= group02,ou=groups,dc= example,dc=com
cn: group02
objectclass: groupOfUniqueNames
objectclass: top
uniquemember: cn= user01,ou=users,dc= example,dc=com
```

### What is the expected behavior?

when i get user01, memberOf attribute does not set
```
dn: cn=user01,ou=users,dc= example,dc=com
cn: user01
createtimestamp: 20220719080619Z
creatorsname: cn=admin,dc=example,dc=com
entrycsn: 20220719080619.489607Z#000000#000#000000
entrydn: cn= user01,ou=users,dc= example,dc=com
entryuuid: 6a597960-9b85-103c-89c9-2986caa8c126
hassubordinates: FALSE
modifiersname: cn=admin,dc= example,dc=com
modifytimestamp: 20220719080619Z
objectclass: inetOrgPerson
objectclass: top
structuralobjectclass: inetOrgPerson
subschemasubentry: cn=Subschema
uid: user01
```

### What do you see instead?

I want to see memberOf attribute also:
```
dn: cn=user01,ou=users,dc= example,dc=com
memberof: cn=group01,ou=groups,dc= example,dc=com
memberof: cn= group02,ou=groups,dc=example,dc=com
...
```

### Additional information

Thanks for your help

**Comment by gongomgra** [comment]
Hi @mohsensamiei,

I reviewed our compilation process and I found out we are not setting the `--enable-memberof` flag in the compilation process of OpenLDAP, which I understand can cause this issue. I will open a new task for our engineering team to update the compilation process and release a new revision of the image. Unfortunately, we can't provide you with an estimation on when this will be updated, but our team will post a new message here once it is finished.

**Comment by zcahana** [comment]
I've just came across this issue myself as well.
As an interim workaround until the memberOf support is compiled into the image, is there some way to configure the image to enable the memberOf overlay?

**Comment by fmulero** [comment]
You can enable `memberOf` feature with the following example, so there is no need to compile it with a special flag:

```yaml
version: '2'

services:
  openldap:
    image: docker.io/bitnami/openldap:2.6
    ports:
      - '1389:1389'
      - '1636:1636'
    environment:
      - BITNAMI_DEBUG=yes
      - LDAP_ADMIN_USERNAME=admin
      - LDAP_ADMIN_PASSWORD=adminpassword
      - LDAP_ROOT=dc=example,dc=com
      - LDAP_GROUP=group01
      - LDAP_USERS=user01,user02
      - LDAP_PASSWORDS=password1,password2
      - LDAP_EXTRA_SCHEMAS=cosine,inetorgperson,nis,memberof
    volumes:
      - 'openldap_data:/bitnami/openldap'
      - './ldifs/memberof.ldif:/opt/bitnami/openldap/etc/schema/memberof.ldif'

volumes:
  openldap_data:
    driver: local
```

And the content of `ldifs/memberof.ldif`:

```
dn: cn=module,cn=config
cn: module
objectClass: olcModuleList
olcModulePath: /opt/bitnami/openldap/lib/openldap
olcModuleLoad: memberof.so
olcModuleLoad: refint.so

dn: olcOverlay=memberof,olcDatabase={2}mdb,cn=config
objectClass: olcMemberOf
objectClass: olcOverlayConfig
olcOverlay: memberof

dn: olcOverlay=refint,olcDatabase={2}mdb,cn=config
objectClass: olcConfig
objectClass: olcOverlayConfig
objectClass: olcRefintConfig
objectClass: top
olcOverlay: refint
olcRefintAttribute: memberof member manager owner
```

Now if you run the command `ldapsearch -D cn=admin,dc=example,dc=com -w adminpassword -H "ldapi:///" -s base -b cn=user01,ou=users,dc=example,dc=com +` you will see something like this:

```
# extended LDIF
#
# LDAPv3
# base <cn=user01,ou=users,dc=example,dc=com> with scope baseObject
# filter: (objectclass=*)
# requesting: +
#

# user01, users, example.com
dn: cn=user01,ou=users,dc=example,dc=com
structuralObjectClass: inetOrgPerson
entryUUID: 7c16f16c-b3dd-103c-9bfe-e3c7b80ddf65
creatorsName: cn=admin,dc=example,dc=com
createTimestamp: 20220819073712Z
entryCSN: 20220819073712.866203Z#000000#000#000000
modifyTimestamp: 20220819073712Z
memberOf: cn=group01,ou=users,dc=example,dc=com
modifiersName: cn=admin,dc=example,dc=com
entryDN: cn=user01,ou=users,dc=example,dc=com
subschemaSubentry: cn=Subschema
hasSubordinates: FALSE

# search result
search: 2
result: 0 Success

# numResponses: 2
# numEntries: 1
```

**Comment by xuzheng0017** [comment]
If the type of group is groupOfUniqueNames, try this config:
```dn: cn=module,cn=config
cn: module
objectClass: olcModuleList
olcModulePath: /opt/bitnami/openldap/lib/openldap
olcModuleLoad: memberof.so
olcModuleLoad: refint.so

dn: olcOverlay=memberof,olcDatabase={2}mdb,cn=config
objectClass: olcConfig
objectClass: olcMemberOf
objectClass: olcOverlayConfig
objectClass: top
olcOverlay: memberof
olcMemberOfDangling: ignore
olcMemberOfRefInt: TRUE
olcMemberOfGroupOC: groupOfUniqueNames
olcMemberOfMemberAD: uniqueMember
olcMemberOfMemberOfAD: memberOf

dn: olcOverlay=refint,olcDatabase={2}mdb,cn=config
objectClass: olcConfig
objectClass: olcOverlayConfig
objectClass: olcRefintConfig
objectClass: top
olcOverlay: refint
olcRefintAttribute: memberof uniqueMember manager owner

**Comment by robigan** [comment]
It isn't working for me either

**Comment by robigan** [comment]
Nvm, somehow now it wishes to work

**Comment by thomas-mc-work** [comment]
Thanks, @fmulero, your example is working for me! But sadly not the one from @xuzheng0017. I'm working in a fresh instance, so no side effects to impair the result.

**Comment by Dzordzu** [comment]
For onlookers:

If you want to use it with `groupOfNames`, go on with @fmulero config

In case of  `groupOfUnqiueNames` use this

```
dn: cn=module,cn=config
cn: module
objectClass: olcModuleList
olcModulePath: /opt/bitnami/openldap/lib/openldap
olcModuleLoad: memberof.so
olcModuleLoad: refint.so

dn: olcOverlay=memberof,olcDatabase={2}mdb,cn=config
objectClass: olcMemberOf
objectClass: olcOverlayConfig
olcOverlay: memberof
olcMemberOfRefInt: TRUE
olcMemberOfGroupOC: groupOfUniqueNames
olcMemberOfMemberAD: uniqueMember
olcMemberOfMemberOfAD: memberOf

dn: olcOverlay=refint,olcDatabase={2}mdb,cn=config
objectClass: olcConfig
objectClass: olcOverlayConfig
objectClass: olcRefintConfig
objectClass: top
olcOverlay: refint
olcRefintAttribute: memberof uniqueMember manager owner
```

**Comment by wotsyula** [comment]
I'm going to post my workaround in case anyone needs it:

00-modules.ldif

```ldif
dn: cn=module{0},cn=config
changetype: modify
add: olcModuleLoad
olcModuleLoad: memberof
olcModuleLoad: refint
```

01-memberof.ldif

```ldif
dn: olcOverlay=memberof,olcDatabase={2}mdb,cn=config
changetype: add
objectClass: olcConfig
objectClass: olcMemberOf
objectClass: olcOverlayConfig
objectClass: top
olcOverlay: memberof
olcMemberOfDangling: ignore
olcMemberOfRefInt: TRUE
olcMemberOfGroupOC: groupOfNames
olcMemberOfMemberAD: member
olcMemberOfMemberOfAD: memberOf
```

02-refint.ldif

```ldif
dn: olcOverlay=refint,olcDatabase={2}mdb,cn=config
changetype: add
objectClass: olcConfig
objectClass: olcOverlayConfig
objectClass: olcRefintConfig
objectClass: top
olcOverlay: refint
olcRefintAttribute: memberof member manager owner
```
